src/RileyMaze/RMZUnit.js

```diff
@@ -0,0 +1,219 @@
+            return;
+        }
+
+        if (Object.getOwnPropertyNames(scope).every(prop => prop === null))
+            return;
+
+        const errors = this._searchForErrors(scope);
+        if (errors.length > 0) {
+            this._variablesCodeview.setCompileResults(errors);
+            return;
+        }
+
+        this._variablesCodeview.setCompileResults([]);
+
+        // Block the normal notify handler that updates the code view, since we
+        // are propagating updates from the codeview to the GUI. Instead,
+        // connect a temporary handler that lets us know if anything actually
+        // did change.
+        GObject.signal_handler_block(this._model, this._notifyHandler);
+
+        let guiUpdated = false;
+        const tempHandler = this._model.connect('notify', () => {
+            guiUpdated = true;
+        });
+        try {
+            if (scope.robotADirection !== null &&
+                this._model.robotADirection !== scope.robotADirection)
+                this._model.set_property('robotADirection', scope.robotADirection);
+            if (scope.robotBDirection !== null &&
+                this._model.robotBDirection !== scope.robotBDirection)
+                this._model.set_property('robotBDirection', scope.robotBDirection);
+        } finally {
+            this._model.disconnect(tempHandler);
+            GObject.signal_handler_unblock(this._model, this._notifyHandler);
+        }
+
+        if (guiUpdated)
+            SoundServer.getDefault().play('hack-toolbox/update-gui');
+    }
+
+    _errorRecordAtAssignmentLocation(variable, message, value) {
+        const {start, end} = this._variablesCodeview.findAssignmentLocation(variable);
+        return {start, end, message, fixme: String(value)};
+    }
+
+    _searchForErrors(scope) {
+        const errors = [];
+        const PROPS_DEFAULTS = {
+            robotADirection: 'up',
+            robotBDirection: 'up',
+        };
+        Object.entries(PROPS_DEFAULTS).forEach(([prop, defaultValue]) => {
+            if (typeof scope[prop] !== 'string') {
+                errors.push(this._errorRecordAtAssignmentLocation(prop,
+                    `Unknown value ${scope[prop]}" for ${prop}: value must be a string`,
+                    defaultValue));
+            }
+        });
+        return errors;
+    }
+
+    _onNotify() {
+        this.reset();
+    }
+
+    reset() {
+        const oldText = this._variablesCodeview.text;
+        this._regenerateCode();
+        const timeMicrosec = GLib.get_monotonic_time();
+        if (this._updateSoundEnabled && !this._model.inReset &&
+            oldText !== this._variablesCodeview.text &&
+            timeMicrosec - this._lastCodeviewSoundMicrosec >= 100e3) {
+            SoundServer.getDefault().play('hack-toolbox/update-codeview');
+            this._lastCodeviewSoundMicrosec = timeMicrosec;
+        }
+    }
+
+    _regenerateCode() {
+        this._variablesCodeview.text = `
```

Comment:
Currently there's a newline on line 1 of this code block, if you put a `\` after the backquote it would remove the newline and be more consistent with other code views.